translation of Passing Data Deeply with Context (#493)
reactjs/fr.react.dev: src/content/learn/passing-data-deeply-with-context.md

</Section>
```

Mais comment le composant `<Heading>` peut-il connaître le niveau de sa `<Section>` la plus proche ? **Il faudrait pour ça qu'un enfant puisse « demander » une donnée à un niveau supérieur de l'arbre**.

Ce n'est pas possible uniquement avec les props. C'est ici que le contexte entre en jeu. Vous allez faire ça en trois étapes :

1. **Créer** un contexte (vous pouvez l'appeler `LevelContext`, puisque c'est le niveau de l'en-tête).
2. **Utiliser** ce contexte au niveau du composant qui a besoin de la donnée (`Heading` utilisera `LevelContext`).
3. **Fournir** ce contexte depuis le composant qui spéficie la donnée (`Section` fournira `LevelContext`).

Les contextes permettent à un parent — aussi distant soit-il — de fournir des données à l'ensemble de l'arbre en dessous de lui.

<DiagramGroup>

<Diagram name="passing_data_context_close" height={160} width={608} captionPosition="top" alt="Un diagramme avec un arbre à trois composants. Le parent contient une bulle représentant une valeur surlignée en orange qui se projettent jusqu'à ses deux enfants, tous deux surlignées en orange." >

Utiliser le contexte avec des enfants proches

</Diagram>

> <Diagram name="passing_data_context_far" height={430} width={608} captionPosition="top" alt="Un diagramme avec un arbres à dix nœeuds, chacun ayant deux enfants ou moins. Le nœud parent à la racine contient une bulle qui représente une valeur surlignée en orange. La valeur se projette directement vers quatres feuilles et un composant intermédiaire de l'arbre, qui sont tous surlignés en orange. Aucun des autres composants intermédiaires n'est surligné.">

Reviewer: ```suggestion
<Diagram name="passing_data_context_far" height={430} width={608} captionPosition="top" alt="Un diagramme avec un arbres à dix nœeuds, chacun ayant deux enfants ou moins. Le nœud parent à la racine contient une bulle qui représente une valeur surlignée en orange. La valeur est projetée directement vers quatre nœuds feuilles et un composant intermédiaire de l'arbre, qui sont tous surlignés en orange. Aucun des autres composants intermédiaires n'est surligné.">
```

Utiliser le contexte avec des enfants éloignés

</Diagram>

</DiagramGroup>

### Étape 1 : créer le contexte {/*step-1-create-the-context*/}

> Tout d'abord, vous devez créer le contexte. Vous avez besoin de **l'exporter depuis un fichier** pour que vos composants puissent l'utiliser :

Reviewer: ```suggestion
Tout d'abord, il vous faut créer le contexte. Vous devrez **l'exporter depuis un fichier** pour que vos composants puissent l'utiliser :
```

</Sandpack>

Le seul argument à `createContext` est la valeur _par défaut_. Ici, `1` fait référence au niveau d'en-tête le plus élevé, mais vous pouvez passer n'importe quel type de valeur (y compris un objet). Vous verrez l'importance des valeurs par défaut dans la prochaine étape.

### Étape 2 : utiliser le contexte {/*step-2-use-the-context*/}

Importez le Hook `useContext` depuis React, ainsi que votre contexte :

```js
import { useContext } from 'react';
import { LevelContext } from './LevelContext.js';
```

> Pour le moment, le composant `Heading` lit `level` depuis le props :

Reviewer: ```suggestion
Pour le moment, le composant `Heading` lit `level` depuis les props :
```

```js
export default function Heading({ level, children }) {
  // ...
}
```

> Supprimez la prop `level`, et lisez plutôt la valeur depuis le contexte `LevelContext` que vous venez d'importer :

Reviewer: Reformulation parfaite, notamment le placement du "plutôt", bravo 👏

}
```

> `useContext` est un Hook. Tout comme `useState` et `useReducer`, vous ne pouvez appeler un Hook qu'au début d'un composant React (et non pas dans des boucles ou des conditions). **`useContext` indique à React que le composant `Heading` souhaite lire le `LevelContext`.**

Reviewer: ```suggestion
`useContext` est un Hook. Tout comme `useState` et `useReducer`, vous ne pouvez appeler un Hook qu'au niveau racine d'un composant React (et non pas dans des boucles ou des conditions). **`useContext` indique à React que le composant `Heading` souhaite lire le `LevelContext`.**
```

Maintenant que votre composant `Heading` n'a plus besoin de la prop `level`, vous n'avez plus besoin de la passer à `Heading` dans votre JSX :

```js
<Section>
  <Heading level={4}>Sous-sous-section</Heading>
  <Heading level={4}>Sous-sous-section</Heading>
  <Heading level={4}>Sous-sous-section</Heading>
</Section>
```

Mettez à jour le JSX afin que `Section` la reçoive désormais :

```jsx
<Section level={4}>
  <Heading>Sous-sous-section</Heading>
  <Heading>Sous-sous-section</Heading>
  <Heading>Sous-sous-section</Heading>
</Section>
```

> Pour rappel, voici le code que vous essayez de faire marcher :

Reviewer: J'aime bien le passage au présent de narration pour éviter un subjonctif imparfait moche sur "essayer".

</Sandpack>

> Remarquez que cet exemple ne fonctionne pas encore tout à fait. Tous les en-têtes ont le même niveau parce que **même si vous *utilisez* le contexte, vous ne l'avez pas encore *fourni***. React ne sait pas où l'obtenir.

Reviewer: ```suggestion
Remarquez que cet exemple ne fonctionne pas encore tout à fait. Tous les en-têtes ont le même niveau parce que **même si vous *utilisez* le contexte, vous ne l'avez pas encore *fourni***. React ne sait pas où l'obtenir !
```

Putain, j'étais **persuadé** qu'en-tête était féminin, mais non, c'est bien masculin. TIL!

Author: Je dois dire que je le savais, sans être persuadé. J'avais fait la vérification.

> Si vous ne fournissez pas le contexte, React utilisera les valeurs par défaut que vous avez spécifié dans l'étape précédente. Dans cet exemple, vous aviez spécifié `1` comme argument à `createContext`, donc `useContext(LevelContext)` renvoie `1`, transformant tous ces en-têtes en `<h1>`. Corrigeons ce problème en demandant à chaque `Section` de fournir son propre contexte.

Reviewer: ```suggestion
Si vous ne fournissez pas le contexte, React utilisera la valeur par défaut que vous avez spécifiée dans l'étape précédente. Dans cet exemple, vous aviez spécifié `1` comme argument à `createContext`, donc `useContext(LevelContext)` renvoie `1`, transformant tous ces en-têtes en `<h1>`. Corrigeons ce problème en demandant à chaque `Section` de fournir son propre contexte.
```

### Étape 3 : fournir le contexte {/*step-3-provide-the-context*/}

> Le composant `Section` effectue actuellement le rendu de ses enfants ainsi :

Reviewer: ```suggestion
Le composant `Section` effectue actuellement le rendu de ses enfants comme suit :
```

</Sandpack>

> Le résultat est le même qu'avec le code d'origine, mais vous n'avez pas eu besoin de transmettre la prop `level` à chaque composant `Heading`. Au lieu de ça, il « détermine » son niveau d'en-tête en interrogeant la `Section` la plus proche :

Reviewer: ```suggestion
Le résultat est le même qu'avec le code d'origine, mais vous n'avez plus besoin de transmettre la prop `level` à chaque composant `Heading`. Au lieu de ça, il « détermine » son niveau d'en-tête en interrogeant la `Section` la plus proche :
```

Léger écart avec la VO mais plus lisible je trouve.

1. Vous passez une prop `level` à la `<Section>`.
2. `Section` enrobe ses enfants dans un `<LevelContext.Provider value={level}>`.
3. `Heading` demande la valeur la plus proche de `LevelContext` avec `useContext(LevelContext)`.

## Utiliser et fournir le contexte depuis le même composant {/*using-and-providing-context-from-the-same-component*/}

> Pour l'instant, vous devez toujours spécifier manuellement le `level` de chaque section :

Reviewer: ```suggestion
À ce stade, vous devez quand même spécifier manuellement le `level` de chaque section :
```

          ...
```

> Puisque le contexte vous permet de lire une information à partir d'un composant plus haut, chaque `Section` pourrait lire le `level` de la `Section` supérieure, puis transmettre automatiquement `level + 1` en dessous de lui. Voici comment faire :

Reviewer: ```suggestion
Puisque le contexte vous permet de lire une information à partir d'un composant plus haut, chaque `Section` pourrait lire le `level` de la `Section` supérieure, puis transmettre automatiquement `level + 1` en dessous d'elle. Voici comment faire :
```

</Sandpack>

Désormais, `Heading` et `Section` lisent le `LevelContext` pour déterminer à quelle « profondeur » ils se trouvent. `Section` enrobe ses enfants dans un `LevelContext` pour spécifier que tout son contenu est à un niveau « plus profond ».

<Note>

> Cet exemple utilise des niveaux d'en-têtes parce qu'ils montrent visuellement à quel point les composants imbriqués peuvent surcharger le contexte. Mais le contexte est aussi utile dans bien d'autres cas. Vous pouvez transmettre n'importe quelle information nécessaire à tout le sous-arbre : le thème de couleur courant, l'utilisateur connecté, et ainsi de suite.

Reviewer: ```suggestion
Cet exemple utilise des niveaux d'en-têtes parce qu'ils montrent visuellement comment des composants imbriqués peuvent surcharger le contexte. Mais le contexte est tout aussi utile dans bien d'autres cas. Vous pouvez transmettre n'importe quelle information nécessaire à tout le sous-arbre : le thème de couleurs actif, l'utilisateur connecté, et ainsi de suite.
```

</Note>

> ## Le contexte passe par des composants intermédiaires {/*context-passes-through-intermediate-components*/}

Reviewer: ```suggestion
## Le contexte traverse les composants intermédiaires {/*context-passes-through-intermediate-components*/}
```

Léger glissement sémantique

> Vous pouvez insérer autant de composants que vous voulez entre le composant qui fournit le contexte et celui qui l'utilise. Ça inclut aussi bien les composants intégrés comme `<div>` que ceux que vous pouvez créer vous-même.

Reviewer: ```suggestion
Vous pouvez insérer autant de composants que vous voulez entre le composant qui fournit le contexte et celui qui l'utilise. Ça inclut aussi bien les composants natifs comme `<div>` que ceux que vous pourriez créer vous-même.
```

function RecentPosts() {
  return (
    <Section>
      <Heading>Billets récents</Heading>
      <Post
        title="Les saveurs de Lisbonne"
        body="...et ses pastéis de nata !"
      />
      <Post
        title="Buenos Aires au rythme du tango"

>         body="J'ai adoré !"

Reviewer: 👍

</Sandpack>

> Vous n'avez rien eu à faire pour que ça marche. Une `Section` spécifie le contexte pour l'arbre qu'elle contient, vous pouvez donc y insérer un `<Heading>` n'importe où et il aura une taille correcte. Essayez donc dans le bac à sable ci-dessus.

Reviewer: ```suggestion
Vous n'avez rien eu à faire pour que ça marche. Une `Section` spécifie le contexte pour l'arbre qu'elle contient, vous pouvez donc y insérer un `<Heading>` n'importe où et il aura le niveau correct. Essayez donc dans le bac à sable ci-dessus.
```

**Le contexte vous permet d'écrire des composants qui « s'adaptent à leur environnement » et s'affichent différemment en fonction de _l'endroit_ (autrement dit _dans quel contexte_) ils sont rendus.**

> La façon dont les contextes fonctionnent peut vous rappeler [l'héritage des propriétés CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/inheritance). En CSS, vous pouvez spécifier `color: blue` pour un `<div>` et n'importe quel nœud du DOM qu'il contient, aussi loin soit-il, héritera de cette couleur, à moins qu'un nœud intermédiaire ne surcharge ça avec `color: green`. De même, dans React, la seule façon de surcharger un contexte venant d'en haut est d'enrober les enfants dans un fournisseur de contexte avec une valeur différente.

Reviewer: ```suggestion
La façon dont les contextes fonctionnent peut vous rappeler [l'héritage des valeurs de propriétés en CSS](https://developer.mozilla.org/fr/docs/Web/CSS/inheritance). En CSS, vous pouvez spécifier `color: blue` pour un `<div>` et n'importe quel nœud du DOM qu'il contient, aussi profond soit-il, héritera de cette couleur, à moins qu'un nœud intermédiaire ne surcharge ça avec `color: green`. C'est pareil avec React : la seule façon de surcharger un contexte venant d'en haut est d'enrober les enfants dans un fournisseur de contexte avec une valeur différente.
```

> En CSS, des propriétés distinctes comme `color` et `background-color` ne sont pas remplacées les unes par les autres. Vous pouvez définir la `color` de toutes les `<div>` en rouge sans impacter le `background-color`. D'une façon similaire, **des contextes React distincts ne se remplacent pas les uns les autres**. Chaque contexte que vous créez avec `createContext()` est complétement isolé des autres et lie les composants qui utilisent et fournissent ce contexte *particulier*. Un composant peut utiliser et fournir différents contextes sans problème.

Reviewer: ```suggestion
En CSS, des propriétés distinctes comme `color` et `background-color` ne sont pas remplacées les unes par les autres. Vous pouvez définir la `color` de toutes les `<div>` en rouge sans impacter la `background-color`. De la même façon, **des contextes React distincts ne s'écrasent pas les uns les autres**. Chaque contexte que vous créez avec `createContext()` est complétement isolé des autres et lie les composants qui utilisent et fournissent ce contexte *particulier*. Un composant peut utiliser et fournir différents contextes sans problème.
```

## Avant d'utiliser un contexte {/*before-you-use-context*/}

> Il est très tentant d'utiliser des contextes ! Toutefois, il est aussi très facile d'en abuser. **Ce n'est pas parce que vous devez descendre des props sur plusieurs niveaux que vous devez mettre ces informations dans un contexte**.

Reviewer: ```suggestion
Il est très tentant d'utiliser des contextes ! Toutefois, il est aussi très facile d'en abuser. **Ce n'est pas parce que vous devez propager des props sur plusieurs niveaux que vous devez mettre ces informations dans un contexte**.
```

Quasi parfait 😉

Voici certaines alternatives à considérer avant d'utiliser un contexte :

> 1. **Commencez par [passer les props](/learn/passing-props-to-a-component)**. Si vos composants ne sont pas triviaux, il est courant de passer une douzaine de props à travers une douzaine de composants. Ça peut sembler fastidieux, mais ça permet de savoir clairement quels composants utilisent quelles données ! La personne chargée de la maintenance de votre code sera ravie que vous ayez rendu le flux de données explicite grâce aux props.
> 2. **Extrayez des composants et [passez-leur du JSX dans les `children`](/learn/passing-props-to-a-component#passing-jsx-as-children)**. Si vous passez des données à travers plusieurs couches de composants qui ne les utilisent pas (et ne font que les transmettre plus bas), ça signifie souvent que vous avez oublié d'extraire certains composants en cours de route. Par exemple, vous pouvez passer certaines données en props comme `posts` à des composants visuels qui ne les utilisent pas directement, tel que `<Layout posts={posts} />`. À la place, faites en sorte que `Layout` prenne `children` en tant que prop, puis faites le rendu de `<Layout><Posts posts={posts} /></Layout>`. Ça réduit le nombre de couches entre les composants qui spécifient la donnée et ceux qui l'utilisent.

Reviewer: ```suggestion
1. **Commencez par [passer les props](/learn/passing-props-to-a-component).** Si vos composants ne sont pas triviaux, il est courant de passer une douzaine de props à travers une douzaine de composants. Ça peut sembler fastidieux, mais ça permet de savoir clairement quels composants utilisent quelles données ! La personne chargée de la maintenance de votre code sera ravie que vous ayez rendu le flux de données explicite grâce aux props.
2. **Extrayez des composants et [passez-leur du JSX dans les `children`](/learn/passing-props-to-a-component#passing-jsx-as-children).** Si vous passez des données à travers plusieurs couches de composants qui ne les utilisent pas (et ne font que les transmettre plus bas), ça signifie souvent que vous avez oublié d'extraire certains composants en cours de route. Par exemple, vous pouvez passer certaines données en props comme `posts` à des composants visuels qui ne les utilisent pas directement, tel que `<Layout posts={posts} />`. Préférez faire en sorte que `Layout` prenne `children` en tant que prop, puis faites le rendu de `<Layout><Posts posts={posts} /></Layout>`. Ça réduit le nombre de couches entre les composants qui spécifient la donnée et ceux qui l'utilisent.
```

> Si aucune de ces approches ne vous convient, envisagez le contexte.

Reviewer: ```suggestion
Si aucune de ces approches ne vous convient, envisagez un contexte.
```

> ## Cas d'utilisation du contexte {/*use-cases-for-context*/}

Reviewer: ```suggestion
## Cas d'utilisation des contextes {/*use-cases-for-context*/}
```

Insistons sur le fait qu'il en existe plusieurs, tous cloisonnés.

* **Thème :** si votre appli permet à l'utilisateur d'en changer l'apparence (comme le mode sombre), vous pouvez mettre un fournisseur de contexte tout en haut de votre appli et utiliser ce contexte dans les composants qui ont besoin d'ajuster leur aspect.
* **Compte utilisateur :** de nombreux composants peuvent avoir besoin de connaître l'utilisateur actuellement connecté. Le fait de le placer dans le contexte en facilite la lecture depuis n'importe quel endroit de l'arbre. Certaines applis vous permettent d'utiliser plusieurs comptes simultanément (par exemple pour laisser un commentaire avec un utilisateur différent). Dans ce cas, il peut être pratique d'enrober une partie de l'interface utilisateur dans un fournisseur avec un compte utilisateur différent.

> * **Routage :** la plupart des solutions de routage utilise un contexte en interne pour conserver la route actuelle. C'est ainsi que chaque lien « sait » s'il est actif ou non. Si vous construisez votre propre routeur, vous serez peut-être tenté de faire de même.

Reviewer: ```suggestion
* **Routage :** la plupart des solutions de routage utilise un contexte en interne pour conserver la route actuelle. C'est ainsi que chaque lien « sait » s'il est actif ou non. Si vous construisez votre propre routeur, vous serez peut-être tenté·e de faire de même.
```

> * **Gestion d'état :** au fur et à mesure que votre appli grandit, vous pouvez vous retrouver avec de nombreux états proches de la racine de votre appli. De nombreux composants distants pourraient vouloir les changer. Il est courant [d'utiliser un réducteur avec un contexte](/learn/scaling-up-with-reducer-and-context) pour gérer des états complexes et les transmettre à des composants distants sans trop de problèmes.

Reviewer: ```suggestion
* **Gestion d'état :** au fur et à mesure que votre appli grandit, vous pouvez vous retrouver avec de nombreux états proches de la racine de votre appli. De nombreux composants distants pourraient vouloir les changer. Il est courant [d'utiliser un réducteur avec un contexte](/learn/scaling-up-with-reducer-and-context) pour gérer des états complexes et les transmettre à des composants distants sans trop galérer.
```

"problèmes" est trop négatif et semble parler de bugs. Si "galérer" te semble vraiment trop familier, "sans trop de cérémonie" est une bonne alternative.

Le contexte ne se limite pas aux valeurs statiques. Si vous passez une valeur différente au prochain rendu, React mettra à jour tous les composants descendants qui le lisent ! C'est pourquoi le contexte est souvent utilisé en combinaison avec l'état.

D'une manière générale, si certaines informations sont nécessaires pour des composants distants dans différentes parties de l'arborescence, alors c'est une bonne indication que le contexte vous sera utile.

<Recap>

* Le contexte permet à un composant de fournir certaines informations à l'ensemble de l'arbre situé en dessous de lui.
* Pour transmettre un contexte :

>   1. Créez et exportez-le avec `export const MyContext = createContext(defaultValue)`.

Reviewer: ```suggestion
  1. Créez-le et exportez-le avec `export const MyContext = createContext(defaultValue)`.
```

>   2. Passez-le au Hook `useContext(MyContext)` pour que n'importe quel composant enfant puisse le lire, aussi distant soit-il.

Reviewer: ```suggestion
  2. Passez-le au Hook `useContext(MyContext)` depuis n'importe quel composant enfant pour pouvoir le lire, aussi profondément imbriqué soit-il.
```

  3. Enrobez les enfants dans un `<MyContext.Provider value={...}>` pour le fournir depuis un parent.
* Le contexte traverse tous les composants intermédiaires.
* Le contexte vous permet d'écrire des composants qui « s'adaptent à leur environnement ».
* Avant d'utiliser un contexte, essayez de passer par les props ou en mettant du JSX dans les `children`.

</Recap>

<Challenges>

#### Remplacer la percolation de props par un contexte {/*replace-prop-drilling-with-context*/}

Dans cet exemple, le fait d'activer la case à cocher change la prop `imageSize` passée à chaque `<PlaceImage>`. L'état de cette case à cocher est conservé dans le composant racine `App`, mais chaque `<PlaceImage>` doit en être informé.

> Pour l'instant, `App` transmet `imageSize` à `List`, qui la transmet ensuite à chaque `Place` qui transmettent enfin à `PlaceImage`. Supprimez la prop `imageSize`, et donnez-la directement à `PlaceImage` depuis le composant `App`.

Reviewer: ```suggestion
Pour l'instant, `App` transmet `imageSize` à `List`, qui la transmet ensuite à chaque `Place` qui transmettent enfin à `PlaceImage`. Supprimez la prop `imageSize` pour la fournir directement à `PlaceImage` depuis le composant `App`.
```

```js data.js
export const places = [{
  id: 0,
  name: 'Bo-Kaap à Cape Town, Afrique du Sud',

>   description: 'la tradition de choisir des couleurs vives pour les maisons a commencé à la fin du XXe siècle.',

Reviewer: ```suggestion
  description: 'La tradition de choisir des couleurs vives pour les maisons remonte à la fin du XXe siècle.',
```

  imageId: 'K9HVAGH'
}, {
  id: 1, 
  name: 'Rainbow Village à Taichung, Taiwan',

>   description: 'pour sauver les maisons de la démolition, Huang Yung-Fu, un résident de la région, a peint 1200 maisons en 1924.',

Reviewer: ```suggestion
  description: 'pour sauver les maisons de la démolition, Huang Yung-Fu, un résident de la région, a peint l'ensemble des 1 200 maisons en 1924.',
```

  imageId: '9EAYZrt'
}, {
  id: 2, 
  name: 'Macromural de Pachuca, Mexique',

>   description: "l'une des plus grandes peintures murales du monde recouvre les maisons d'un quartier à flanc de colline.",

Reviewer: ```suggestion
  description: "L'une des plus grandes fresques murales du monde recouvre les maisons d'un quartier à flanc de colline.",
```

  imageId: 'DgXHVwu'
}, {
  id: 3, 
  name: 'Selarón Staircase à Rio de Janeiro, Brésil',

>   description: 'ce monument a été créé par Jorge Selarón, un artiste chilien, en guise d’« hommage au people du Brésil ».',

Reviewer: ```suggestion
  description: 'Ce monument a été créé par Jorge Selarón, un artiste d'origine chilienne, en guise d’« hommage au people du Brésil ».',
```

  imageId: 'aeO3rpI'
}, {
  id: 4, 
  name: 'Burano, Italie',

>   description: 'les maisons sont peintes selon un système de couleurs spécifique datant du XIVe siècle.',

Reviewer: ```suggestion
  description: 'Les maisons sont peintes selon un système de couleurs spécifique datant du XVIe siècle.',
```

  imageId: 'kxsph5C'
}, {
  id: 5, 
  name: 'Chefchaouen, Maroc',

>   description: "plusieurs théories expliquent pourquoi les maisons sont peintes en bleu, notamment parce que cette couleur repousse les moustiques ou qu'elle symbolise le ciel et le paradis.",

Reviewer: ```suggestion
  description: "Plusieurs théories expliquent pourquoi les maisons sont peintes en bleu, notamment parce que cette couleur repousserait les moustiques ou qu'elle symboliserait le ciel et le paradis.",
```

  imageId: 'rTqKo46'
}, {
  id: 6,
  name: 'Gamcheon Culture Village à Busan, Corée du Sud',

>   description: 'en 2009, le village a été converti en centre culturel en peignant les maisons et en accueillant des exhibitions et des installations artistiques.',

Reviewer: ```suggestion
  description: 'En 2009, le village a été converti en carrefour culturel en peignant les maisons et en accueillant des expositions et des installations artistiques.',
```

<Solution>

> Supprimez la prop `imageSize` de tous les composants.

Reviewer: ```suggestion
Retirez la prop `imageSize` de tous les composants.
```

> Créez et exportez `ImageSizeContext` depuis `Context.js`. Ensuite, enrobez la `List` dans `<ImageSizeContext.Provider value={imageSize}>` pour faire descendre la valeur, puis `useContext(ImageSizeContext)` pour la lire dans `PlaceImage`:

Reviewer: ```suggestion
Créez et exportez `ImageSizeContext` depuis `Context.js`. Ensuite, enrobez la `List` dans `<ImageSizeContext.Provider value={imageSize}>` pour propager la valeur vers le bas, puis `useContext(ImageSizeContext)` pour la lire dans `PlaceImage` :
```

```js data.js
export const places = [{
  id: 0,
  name: 'Bo-Kaap à Cape Town, Afrique du Sud',

>   description: 'la tradition de choisir des couleurs vives pour les maisons a commencé à la fin du XXe siècle.',

Reviewer: Merci de reporter ici et plus bas les ajustements de trads pour les légendes ! 😁

Author: Fait. J'ai aussi changer des `'`en `"` là où il fallait.

</Sandpack>

> Remarquez comment les composants intermédiaires n'ont plus besoin de transmettre `imageSize`.

Reviewer: ```suggestion
Remarquez comme les composants intermédiaires n'ont plus besoin de transmettre `imageSize`.
```
